spec/booking_manager_spec.rb

```diff
@@ -0,0 +1,333 @@
+        block_name: "Munster - Addams Wedding",
+        block_discount: 0.08,
+        number_of_rooms_to_block: 3
+      }
+      @input2 = {
+        check_in_date: @date,
+        check_out_date: @date2,
+        block_name: "Munster - Addams Wedding",
+        block_discount: 0.08,
+        number_of_rooms_to_block: 5
+      }
+      @input3 = {
+        check_in_date: @date,
+        check_out_date: @date2,
+        block_name: "Munster - Addams Wedding",
+        block_discount: 0.08,
+        number_of_rooms_to_block: 6
+      }
+    end
+
+    it 'creates an array of block room reservations and adds it an array of all block room reservations ' do
+      @block = @hotel.create_block(@input)
+      expect(@hotel.block_reservations.length).must_equal 1
+      @hotel.block_reservations[0].each do |reservation|
+        expect(reservation).must_be_kind_of Hotel::Block_Room
+        expect(reservation.block_name).must_equal "Munster - Addams Wedding"
+        expect(reservation.check_in_date).must_equal @date
+        expect(reservation.nights_of_stay.length).must_equal 3
+      end
+    end
+
+    it 'raises an Exception if you try to create a block of the same dates' do
+      @hotel.create_block(@input2)
+      @hotel.create_block(@input2)
+      @hotel.create_block(@input2)
+      @hotel.create_block(@input2)
+      expect{@hotel.create_block(@input2)}.must_raise ArgumentError
+    end
+
+    it 'raises an Exception if you try to create a block of more than 5 rooms' do
+      expect{@hotel.create_block(@input3)}.must_raise ArgumentError
+    end
+
+  end
+
+  describe 'check_block_status method' do
+    before do
+      @hotel = Hotel::Booking_Manager.new
+      @date = Date.new(2028,8,8)
+      @date2 = Date.new(2028,8,11)
+      @input = {
+        check_in_date: @date,
+        check_out_date: @date2,
+        block_name: "Munster - Addams Wedding",
+        block_discount: 0.08,
+        number_of_rooms_to_block: 3
+      }
+      @input2 = "Munster - Addams Wedding"
+      @input3 = {block_name: "Munster - Addams Wedding", name: "Cousin IT"}
+      @block = @hotel.create_block(@input)
+    end
+
+    it 'takes a name of a block and returns an array of rooms in that block have that have a status of :AVAILABLE' do
+      @check = @hotel.check_block_status(@input2)
+      expect(@check).must_be_kind_of Array
+      expect(@check.length).must_equal 3
+      expect(@check.first.block_reservation_status).must_equal :AVAILABLE
+    end
+
+    it 'returns a smaller array if one of the block rooms gets booked' do
+      @hotel.reserve_room_in_block(@input3)
+      @check2 = @hotel.check_block_status(@input2)
+      expect(@check2.length).must_equal 2
+    end
+
+    it 'returns an error if all rooms in block booked' do
+      @hotel.reserve_room_in_block(@input3)
+      @hotel.reserve_room_in_block(@input3)
+      @hotel.reserve_room_in_block(@input3)
+      expect{ @hotel.check_block_status(@input2) }.must_raise StandardError
```

Comment:
I feel like I would want this error to be raised if I tried to call `@hotel.reserve_room_in_block(@input3)` again instead of `check_block_status`?

I think this raises questions for me about what the responsibility of the method `check_block_status` is